### PR TITLE
Replace all logging methods with klogv2, if they do not use klog

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Create a 'NodeFeatureDiscoveryLogger' object that is a wrapper
around klog itself. By using a wrapper and only using the new
'r.Log.*", we can ensure that the logging libraries used are
consistent across all files, and we can also ensure that it is easy
to change the logging library from klog to another logging library,
if that is desired in the future.

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>
Co-Authored-by: Courtney Pacheco <cpacheco@redhat.com>

### CODE REVIEW

File: controllers/nodefeaturediscovery_controller.go [hunk 1]
```diff
@@ -133,25 +129,21 @@ func validateUpdateEvent(e *event.UpdateEvent) bool {
...
-	// request likely needs to be requeued.
+	// Error reading the object - requeue the request.
 	if err != nil {
 		// handle deletion of resource
 		if k8serrors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup
-			// logic use finalizers. Return and don't requeue.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:146]: nit: why change this, looks like it's going worse (wrapping) 😄

File: controllers/nodefeaturediscovery_controller.go [hunk 2]
```diff
@@ -133,25 +129,21 @@ func validateUpdateEvent(e *event.UpdateEvent) bool {
 // +kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheusrules,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims
-// to move the current state of the cluster closer to the desired state.
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:133]: nit: unrelated, unnecessary

File: controllers/nodefeaturediscovery_controller.go [hunk 3]
```diff
@@ -133,25 +129,21 @@ func validateUpdateEvent(e *event.UpdateEvent) bool {
...
 
 	// Fetch the NodeFeatureDiscovery instance on the cluster
 	r.Log.Info("Fetch the NodeFeatureDiscovery instance")
 	instance := &nfdv1.NodeFeatureDiscovery{}
 	err := r.Get(ctx, req.NamespacedName, instance)
-
-	// If an error occurs because "r.Get" cannot get the NFD instance
-	// (e.g., due to timeouts, aborts, etc. defined by ctx), the
-	// request likely needs to be requeued.
+	// Error reading the object - requeue the request.
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:140]: nit: unrelated, possibly a separate PR if you want to update the comments(?)

File: controllers/nodefeaturediscovery_controller.go [hunk 4]
```diff
@@ -19,60 +19,58 @@ package controllers
...
 
-// SetupWithManager sets up the controller with a specified manager responsible for
-// initializing shared dependencies (like caches and clients)
+// SetupWithManager sets up the controller with the Manager.
 func (r *NodeFeatureDiscoveryReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
-	// The predicate package is used by the controller to filter events before
-	// they are sent to event handlers. Use it to initate the reconcile loop only
-	// on a spec change of the runtime object.
+	// we want to initate reconcile loop only on spec change of the object
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:70]: nit: unrelated

File: controllers/nodefeaturediscovery_controller.go [hunk 5]
```diff
@@ -19,60 +19,58 @@ package controllers
...
+	client.Client
+	Log       NodeFeatureDiscoveryLogger
+	Scheme    *runtime.Scheme
+	Recorder  record.EventRecorder
+	AssetsDir string
 }
 
-// SetupWithManager sets up the controller with a specified manager responsible for
-// initializing shared dependencies (like caches and clients)
+// SetupWithManager sets up the controller with the Manager.
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:67]: unrelated(?)

File: controllers/nodefeaturediscovery_controller.go [hunk 6]
```diff
@@ -19,60 +19,58 @@ package controllers
...
-
 // nfd is an NFD object that will be used to initialize the NFD operator
 var nfd NFD
 
-// NodeFeatureDiscoveryReconciler reconciles a NodeFeatureDiscovery object
-type NodeFeatureDiscoveryReconciler struct {
+// NodeFeatureDiscoveryLogger is a dummy logger struct that is used with
+// the NodeFeatureDiscoveryReconciler to initiate a logger.
+type NodeFeatureDiscoveryLogger struct {
+}
```
Comment (marquiz) [inline controllers/nodefeaturediscovery_controller.go:44]: I don't quite get the usefulness of this? Why not just add logger methods directly to `NodeFeatureDiscoveryReconciler`? As it doesn't really implement any logger interface (known to me) I'd suggest to scrap the logging methods and do simply`s/logger/klog/`. WDYT?

**Review by marquiz — COMMENTED** [review]
Thanks, "one logger everywhere" is good. However, to me, looks overly complicated without much benefits. I suggest to drop the separate logging methods (besides the extra compexity they fuzz the filenames/line numbers of log messages, I think) and use klog directly everywhere

**Comment by ArangoGutierrez** [comment]
In favor of https://github.com/kubernetes-sigs/node-feature-discovery-operator/pull/93
/close
